### ISSUE TITLE
Add equivalent of `getnameinfo`

### ISSUE THREAD

**Issue by badeend** [WebAssembly/wasi-sockets#3]
Should the inverse of `resolve_addresses` be part of this proposal? Eg `resolve_name` based on `getnameinfo`.

Although they would be each other's mirror image, their use cases are very different. `resolve_addresses` is used by almost every application that is connected to the internet, while its inverse `resolve_name` is relatively niche.

- Pro: symmetry
- Con: Not part of the MVP spirit. Can be added later.

**Comment by sunfishcode** [comment]
Another Pro is that it improves compatibility with existing code that uses this feature. But it still seems relatively niche.

What if we include it in the proposal for now, and see how it goes? If we encounter complications, we can remove it. But if it's traighforward, we can just include it, and save ourselves the extra work of discovering applications in the future that eventually need it, and then adding it for them.

**Comment by badeend** [comment]
If we were to include `resolve_name`/`getnameinfo`, does that also open the door for inclusion of `gethostname` ?

**Comment by sunfishcode** [comment]
Yeah, since we have `getsockname` and if we add `resolve_name`, `gethostname` isn't all that different. We should still associated it with a handle though, to avoid having an implied ambient "host".

**Comment by badeend** [comment]
Not sure I see the relationship. Despite their unfortunate names, `getsockname` & `getpeername` don't return names but socket addresses. `gethostname` returns the name of the current computer.

> We should still associated it with a handle though, to avoid having an implied ambient "host".

Ofcourse 😄 Part of the "network" capability or part of some kind of "host info" capability?

**Comment by sunfishcode** [comment]
Yeah; `getsockname` returns an address, and then you could use `getaddrinfo` to get the name. That's not exactly the same as `gethostname`, but big picture, there doesn't seem to be a fundamental reason not to include `gethostname`.

> Ofcourse smile Part of the "network" capability or part of some kind of "host info" capability?

This isn't necessarily definitive, Linux has separate Network and UTS namespaces, so maybe it makes sense to make this a separate "host info" capability.
